gityghasemi.py

```diff
@@ -0,0 +1,123 @@
+cell_ind=0
+cells=[0]*100000
+i=0
+j=0
+YOUR_NAME=""
+
+# positions of  '[' and ']' 
+loop_pos = {}
+
+# stack of open brackets
+stack = []
+
+bf_code=">+++++++[<++++++++++>u-]<+.>>+++++++++[<+++++++++>-]<+.>>+++dgdfgada++++++++++++++++++[<++++>-]<.>>+++++++++++[<++++++++>-]<+.>++++++++++++++++++++++++++++++++.>>+++++++[<++++++++++>-]<+.>>++++++++[<+++++++++>-]<.>>+++++++++++++[<+++++>-]<.>>+++++++++++++++++++++[<++++>-]<-.>>+++++++++++++++++++++++[<+++>-]<.>>+++++++++++[<++++dasdasfdsg+++>-]<.>>++++++++[<+++++++++>-]<+."
+
+while j<len(bf_code):
+	# push the location of open brackets onto the stack
+    if bf_code[j]=="[":
+        stack.append(j)
+        
+	# when a closing bracket is found, get the starting position and store the start/end pair
+    if bf_code[j]== "]":
+        if len(stack)==0:
+            raise Exception('Loop end when no loops open at location %i' % j)
+        start = stack.pop()
+        loop_pos[start]=j
+        loop_pos[j]=start
+    j+=1
+
+	# adding pointer
+def addpointer(): 
+        global cell_ind,i
+        cell_ind=cell_ind+1
+        if cell_ind>len(cells):
+            raise Exception('error:overflow pointer')
+        i=i+1

```

Comment:
Since you increase the i variable in all the functions, I think it is better to move all of them out of function and put it after calling the function.

Reply:
you're right.I will change it